instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/middlewares/subscriber.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'command_serializer'
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      module Middlewares
+        # Event handler class for Mongo Ruby driver
+        class Subscriber
+          THREAD_KEY = :__opentelemetry_mongo_spans__
+
+          def started(event) # rubocop:disable Metrics/AbcSize
+            # start a trace and store it in the current thread; using the `operation_id`
+            # is safe since it's a unique id used to link events together. Also only one
+            # thread is involved in this execution so thread-local storage should be safe. Reference:
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+            # TODO: decide naming convention
+            name = 'mongo.cmd'
+            # name = "#{event.database_name}.#{event.command_name}"
+            span = tracer.start_span(name, kind: :client)
```

Comment:
```suggestion
            collection = get_collection(event.command)
            span = tracer.start_span("#{collection || ''}.#{event.command_name}", kind: :client)
```
Based on example at https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md#mongodb